### PR TITLE
Use `$()` to return matched string instead of `head:* tail:*`

### PR DESCRIPTION
**Pull request by dmsnell**

### CODE REVIEW

**Comment by aduth** [comment]
I have no context for the motivation or impact of this change.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 node_modules
 build
+.idea
```
Comment (ntwb) [inline .gitignore]: This shouldn't be in the same pull request as a proposed code change, a separate pull request "if any".

This really should be added to your global `.gitignore`, there are too many IDEs and Editors to accomadate, see https://core.trac.wordpress.org/changeset/35254

**Comment by dmsnell** [comment]
> I have no context for the motivation or impact of this change.

@aduth just a little cleanup to a simpler way - shouldn't cause any functional changes

**Comment by nylen** [comment]
From https://pegjs.org/documentation#grammar-syntax-and-semantics:

> **$ expression**
> Try to match the expression. If the match succeeds, return the matched text instead of the match result.

Remove the `.idea` change and :shipit:

**Pull request merged**